tests/unit/test_reservation.py

```diff
@@ -649,6 +657,254 @@ def test___no_reservation_order___get_connections___connections_returned_in_defa
+
+    assert [_get_subset_with_multiplexer(conn) for conn in connections] == [
+        _ConnectionSubset("Pin3", 1, "Dev2", "5", "Mux0", "route6"),
+        _ConnectionSubset("Pin1", 1, "Dev1", "3", "Mux0", "route4"),
+        _ConnectionSubset("Pin4", 1, "Dev3", "7", "Mux0", "route8"),
+        _ConnectionSubset("Pin2", 1, "Dev2", "4", "Mux0", "route5"),
+        _ConnectionSubset("Pin3", 0, "Dev1", "2", "Mux0", "route3"),
+        _ConnectionSubset("Pin1", 0, "Dev0", "0", "Mux0", "route1"),
+        _ConnectionSubset("Pin4", 0, "Dev3", "6", "Mux0", "route7"),
+        _ConnectionSubset("Pin2", 0, "Dev0", "1", "Mux0", "route2"),
+    ]
+
+
+def test___no_reservation_order___get_connections_with_multiplexer___returns_connections_in_default_order(
+    session_management_client: Mock,
+) -> None:
+    grpc_session_infos = _create_grpc_session_infos_for_ordering()
+    grpc_multiplexer_session_infos = create_nimultiplexer_session_infos(1)
+    reservation = MultiSessionReservation(
+        session_management_client, grpc_session_infos, grpc_multiplexer_session_infos
+    )
+
+    connections = reservation.get_connections_with_multiplexer(object, object)
+
+    assert [_get_subset_with_multiplexer(conn) for conn in connections] == [
+        _ConnectionSubset("Pin1", 0, "Dev0", "0", "Mux0", "route1"),
+        _ConnectionSubset("Pin2", 0, "Dev0", "1", "Mux0", "route2"),
+        _ConnectionSubset("Pin3", 0, "Dev1", "2", "Mux0", "route3"),
+        _ConnectionSubset("Pin4", 0, "Dev3", "6", "Mux0", "route7"),
+        _ConnectionSubset("Pin1", 1, "Dev1", "3", "Mux0", "route4"),
+        _ConnectionSubset("Pin2", 1, "Dev2", "4", "Mux0", "route5"),
+        _ConnectionSubset("Pin3", 1, "Dev2", "5", "Mux0", "route6"),
+        _ConnectionSubset("Pin4", 1, "Dev3", "7", "Mux0", "route8"),
+    ]
+
+
+def test____partial_multiplexed_connections___get_connections_with_multiplexer_using_session_type___returns_all_specified_pin_connections(
+    session_management_client: Mock,
+) -> None:
+    grpc_session_infos = create_nifake_session_infos(1)
+    grpc_multiplexer_session_infos = create_nimultiplexer_session_infos(1)
+    grpc_session_infos[0].channel_mappings.add(pin_or_relay_name="Pin1", site=0, channel="0")
+    grpc_session_infos[0].channel_mappings.add(
+        pin_or_relay_name="Pin2",
+        site=0,
+        channel="1",
+        multiplexer_resource_name="Mux0",
+        multiplexer_route="route1",
+    )
+    grpc_session_infos[0].channel_mappings.add(
+        pin_or_relay_name="Pin3",
+        site=1,
+        channel="1",
+        multiplexer_resource_name="Mux0",
+        multiplexer_route="route2",
+    )
+    grpc_session_infos[0].channel_mappings.add(pin_or_relay_name="Pin4", site=0, channel="3")
+    reservation = MultiSessionReservation(
+        session_management_client, grpc_session_infos, grpc_multiplexer_session_infos
+    )
+
+    # TODO: Update the multiplexer_session_type to fake_driver.Session once the
```

Comment:
FYI: You should add a 2nd fake driver for multiplexers so that these tests aren't passing the same type for both parameters.

Reply:
With the decoupling of multiplexer-related tests into a separate file. Now, I've added a `ni_fake_multiplexer_driver`.